File: tests/fixtures/fixture_tauri/src/services/payment.ts
import { invoke } from '@tauri-apps/api/core';
import { Order } from './order';

export async function processPayment(order: Order): Promise<boolean> {
  return await invoke('process_payment', { order });
}

export async function refundPayment(order: Order): Promise<boolean> {
  return await invoke('refund_payment', { order });
}

File: tests/fixtures/fixture_tauri/src-tauri/src/commands/payment.rs
use crate::models::order::Order;
use tauri::command;

#[command]
pub fn process_payment(order: Order) -> Result<bool, String> {
    if order.total() <= 0.0 {
        return Err("Invalid amount".to_string());
    }
    Ok(true)
}

#[command]
pub fn refund_payment(order: Order) -> Result<bool, String> {
    if order.total() <= 0.0 {
        return Err("Cannot refund zero amount".to_string());
    }
    Ok(true)
}
